sql/expression/function/json/json_value.go

```diff
@@ -169,3 +162,24 @@ func (j *JsonValue) String() string {
 	}
 	return fmt.Sprintf("json_value(%s)", strings.Join(parts, ", "))
 }
+
+var InvalidJsonArgument = errors.NewKind("invalid data type for JSON data in argument 1 to function json_value; a JSON string or JSON type is required")
+
+func GetJSONOrCoercibleString(js interface{}) (jsonData interface{}, err error) {
```

Comment:
the naming implies the argument shape but doesn't give expectations of the return value. MapFromJSONOrCoercibleString`? Maybe docstring?

Reply:
It's not necessarily a map, since it could also any other value that can be represented in JSON.

I renamed it to `GetJSONFromWrapperOrCoercibleString` and added a docstring.